src/vm/codeman.cpp

```diff
@@ -2214,7 +2234,15 @@ HeapList* EEJitManager::NewCodeHeap(CodeHeapRequestInfo *pInfo, DomainCodeHeapLi
     size_t reserveSize = requestAndHeadersSize;
     if (reserveSize < minReserveSize)
         reserveSize = minReserveSize;
+
+#if defined(FEATURE_JIT_PITCHING)
+    if (CLRConfig::GetConfigValue(CLRConfig::INTERNAL_JitPitchEnabled) != 0)
+        reserveSize = ALIGN_UP(reserveSize, OS_PAGE_SIZE);
+    else
+        reserveSize = ALIGN_UP(reserveSize, VIRTUAL_ALLOC_RESERVE_GRANULARITY);
+#else
```

Comment:
With HostCodeHeap you probably don't want this. The HostCodeHeap does use VirtualAlloc with MEM_RESERVE so it is going to expect that its heaps aren't smaller than the minimum reservation size.